selfservice/strategy/oidc/error.go

```diff
@@ -17,3 +25,17 @@ var (
 	ErrAPIFlowNotSupported = herodot.ErrBadRequest.WithError("API-based flows are not supported for this method").
 				WithReasonf("Social Sign In and OpenID Connect are only supported for flows initiated using the Browser endpoint.")
 )
+
+func logUpstreamError(l *logrusx.Logger, resp *http.Response) error {
+	if resp.StatusCode == http.StatusOK {
```

Comment:
Why not
```suggestion
	if resp.StatusCode / 100 == 2 {
```
in case the status code is some other 2XX?

Reply:
Good point. I also thought of doing `200 >= x < 300` but decided against it. As far as I could tell from my research, all userinfo endpoints will return 200. Once exception might be a 3xx redirect, however that should be handled be the roundtripper. One problem with accepting all 200 is that we also accept 204 (no content) which would bring back the issue of an empty response body. 204 might be emitted when the service is having a disruption or something. So I think it makes sense to explicitly test for 200. For any service that doesn't return 200 but instead idk 201 (doesn't make sense though imo), we could adjust the logic. WDYT?